lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetSchemaConverter.java

```diff
@@ -135,12 +133,10 @@ else if (decimalType.isShort()) {
             TimestampType timestampType = (TimestampType) type;
 
             if (timestampType.getPrecision() <= 3) {
-                // TODO TIMESTAMP_MILLIS corresponds to timestampType(isAdjustedToUTC=true, MILLIS), while isAdjustedToUTC should likely be false
-                return Types.primitive(PrimitiveType.PrimitiveTypeName.INT64, OPTIONAL).as(TIMESTAMP_MILLIS).named(name);
+                return Types.primitive(PrimitiveType.PrimitiveTypeName.INT64, OPTIONAL).as(LogicalTypeAnnotation.timestampType(false, LogicalTypeAnnotation.TimeUnit.MILLIS)).named(name);
```

Comment:
I'd wrap this line before `.as` and `.named`; same for the one a few lines down.

```java
                 return Types.primitive(PrimitiveType.PrimitiveTypeName.INT64, OPTIONAL)
                         .as(LogicalTypeAnnotation.timestampType(false, LogicalTypeAnnotation.TimeUnit.MILLIS))
                         .named(name);
```

Reply:
I am keeping it identical to the existing line with nanos

Reply:
Ah. That one should probably also be wrapped, but then it's a separate issue.